app/src/main/java/org/wikipedia/talk/TalkTopicActivity.kt

```diff
@@ -323,6 +324,19 @@ class TalkTopicActivity : BaseActivity(), LinkPreviewDialog.Callback {
                 }))
     }
 
+    private fun undoSave() {
+        disposables.add(CsrfTokenClient(pageTitle.wikiSite).token
+            .subscribeOn(Schedulers.io())
+            .flatMap { token -> ServiceFactory.get(pageTitle.wikiSite).postUndoEdit(pageTitle.prefixedText, currentRevision, token) }
```

Comment:
We should not be undoing the `currentRevision`, since it's not guaranteed to be the revision that you submitted. (Someone else might have made an edit immediately after you did.)

We should create a new private field such as `revisionForUndo` and update it directly from the edit response, and then use it for undoing.

Reply:
You are correct! Done!